homeassistant/components/recorder/statistics.py

```diff
@@ -1597,17 +1640,59 @@ def _statistics_during_period_with_session(
         types,
     )
 
-    # Return statistics combined with metadata
-    if period not in ("day", "week", "month"):
-        return result
-
     if period == "day":
-        return _reduce_statistics_per_day(result, types)
+        result = _reduce_statistics_per_day(result, types)
 
     if period == "week":
-        return _reduce_statistics_per_week(result, types)
+        result = _reduce_statistics_per_week(result, types)
+
+    if period == "month":
+        result = _reduce_statistics_per_month(result, types)
 
-    return _reduce_statistics_per_month(result, types)
+    if "change" in _types:
+        drop_sum = "sum" not in _types
+        prev_sums = {}
+        if tmp := _statistics_at_time(
+            session,
+            {metadata[statistic_id][0] for statistic_id in result},
+            table,
+            start_time,
+            {"sum"},
+        ):
+            _metadata = dict(metadata.values())
+            for row in tmp:
+                metadata_by_id = _metadata[row.metadata_id]
+                statistic_id = metadata_by_id["statistic_id"]
+
+                state_unit = unit = metadata_by_id["unit_of_measurement"]
+                if state := hass.states.get(statistic_id):
+                    state_unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
```

Comment:
Would be nice to cover this line

Reply:
Fixed by this commit in parent PR #92823: https://github.com/home-assistant/core/pull/92823/commits/f9811b3bfabd6e5b048cf5a018ed25ac465cffa4